test/test_mps.py

```diff
@@ -3637,6 +3640,70 @@ def rotate_subset(data, dim):
                 # TODO: enable memory format test
                 # self.assertEqual(cpu_result.is_contiguous(), mps_result.is_contiguous())
 
+    # Skip if a test needs more memory than the system has.
+    def _skip_if_exceeds_total_memory(self, required_memory):
+        if total_memory < required_memory:
+            self.skipTest(
+                f"Needs {required_memory / (1024**3):0.01f} GiB RAM, "
+                f"but only {total_memory / (1024**3):0.01f} GiB is available.")
+
+    @parametrize("dtype", MPS_DTYPES)
+    def test_cat_large_tensor(self, dtype):
```

Comment:
Note that it takes about 1 second to run this for one of the dtypes, so we could just test a bare minimum of them

Reply:
Its an important test , I think 1s should be okay